src/momentOfTruth.ts

```diff
@@ -66,189 +67,211 @@ export async function getLinterResult(
+  const changedFileAndTagsMap = await utils.getTagsFromChangedFile(
     await utils.getFilesChangedInPR(pr)
   );
 
+  if (changedFileAndTagsMap.size === 0) {
+    console.log("Since no tag contains changed swagger, skip run lintDiff");
+    return
+  }
+
   console.log("Processing configs:");
   console.log(configsToProcess);
-  createLogFile();
-  console.log(`The results will be logged here: "${logFilepath}".`);
 
-  const errors: { error: Error; old: string; new: string }[] = [];
+  const linter = new LinterRunner(changedFileAndTagsMap, pr);
+  console.log(`The results will be logged here: "${logFilepath}".`);
 
   if (configsToProcess.length > 0 && pr !== undefined) {
-    for (const configFile of configsToProcess) {
-      try {
-        await runTools(configFile, "after");
-      } catch (err) {
-          errors.push({
-            error: err,
-            old: utils.targetHref(
-              utils.getRelativeSwaggerPathToRepo(
-                path.resolve(pr!.workingDir, configFile)
-              )
-            ),
-            new: utils.blobHref(utils.getRelativeSwaggerPathToRepo(configFile)),
-          });
-        }
-    }
-
+    
+    await linter.runTools("after");
+     
     await utils.doOnTargetBranch(pr, async () => {
-      for (const configFile of configsToProcess) {
-        try {
-          await runTools(configFile, "before");
-        } catch (err) {
-          errors.push({
-            error: err,
-            old: utils.targetHref(
-              utils.getRelativeSwaggerPathToRepo(
-                path.resolve(pr!.workingDir, configFile)
-              )
-            ),
-            new: utils.blobHref(utils.getRelativeSwaggerPathToRepo(configFile)),
-          });
-        }
-      }
+      await linter.runTools("before");
     });
   }
 
-  writeContent(JSON.stringify(finalResult, null, 2));
+  store.writeContent(JSON.stringify(linter.getResult(), null, 2));
+
+  console.log("--- Lint Violation Result ----\n");
+  console.log(JSON.stringify(linter.getResult(), null, 2));
 
-  if (errors.length > 0) {
+  if (linter.getError().length > 0) {
     process.exitCode = 1;
     console.log(`LintDiff error log ----`);
-    const errorResult: format.MessageLine = errors.map((it) => ({
+    const errorResult: format.MessageLine = linter.getError().map((it) => ({
       type: "Raw",
       level: "Error",
-      message: it.error.message || "",
+      message: it.message || "",
       time: new Date(),
       extra: {
-        role: "Lint Diff",
-        new: it.new,
-        old: it.old,
+        role: it.type,
+        new: it.readmeUrl,
       },
     }));
 
     console.log("--- Errors of Lint Diff (formated) ----\n");
-    console.log(JSON.stringify(errorResult));
-    fs.appendFileSync("pipe.log", JSON.stringify(errorResult) + "\n");
+    console.log(JSON.stringify(errorResult,undefined,2));
+    fs.writeFileSync("pipe.log", JSON.stringify(errorResult) + "\n");
     throw new Error('Autorest fail');
```

Comment:
Since we already parse the Autorest error, we can remove the 'throw' statement here.